tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -234,7 +234,7 @@ sub prepare_guest_for_sriov_passthrough {
         assert_script_run "virsh dumpxml --inactive $vm > $vm.xml";
         script_run "virsh destroy $vm";
 
-        if (is_sle('>=15-SP2') && is_kvm_host) {
+        if ((is_sle('>=15-SP2') || is_tumbleweed) && is_kvm_host) {
```

Comment:
Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
@okurz FYI. According to latest discussion result in [2021-07-05 QA&VT Meeting notes](https://confluence.suse.com/pages/viewpage.action?pageId=777912389) for factory first with Antoine, Claudio, Marita and Calen, difference between tumbleweed and sles is not tiny and has no way to eliminate, so the pkgs that are tested on tumbleweed are different from those on sles. As a result, bugs that are found on tumbleweed are not necessarily bugs on sles, so the value of testing tumbleweed by qa-sle is much smaller than expected. So we should put only very limited effort on extending tumbleweed testing – focus only on very limited core tests. 
With this, our strategy for openqa virtualization automation code is  more like that the SLE under development is the "default" product, because potentially we can have a list of middle/low priority automation tests that do not run on tumbleweed, but run on sle.
But I agree with your point that any product with differing behavior from the default should have exclude rule and explanations for why.

Reply:
Sure. But You can try to negate the logic check to not explicitly state Tumbleweed as default, rather treat older versions of SLE or Leap the exception based on specific versions. For this it's not relevant what you have discussed in any internal meeting.

Reply:
> Sure. But You can try to negate the logic check to not explicitly state Tumbleweed as default, rather treat older versions of SLE or Leap the exception based on specific versions. For this it's not relevant what you have discussed in any internal meeting.

I was giving reasons why we use sle as default, rather than tumbleweed. Seems it is fine to you?  I do not quite understand your words here...

Reply:
Please just see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style . I hope that makes it clear and explains what we intend to do on the scale of the complete project os-autoinst-distri-opensuse.

Reply:
I see. We should follow this common rule of the whole project that tw is the default product. But we can set our IF conditions to constrain what products the code paths can run on. Thanks!

Reply:
The above is still unsolved. Better try to negate the logic expression to treat Tumbleweed the new default behaviour and everything else the exception

Reply:
Replaced the if statement with unless statement to remove is_tumbleweed.